src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -155,11 +155,24 @@ public boolean hasResultForCorrectionRound(int correctionRound) {
      * removes all automatic results from a submissions result list
      * (do not save it like this in the database, as it could remove the automatic results!)
      */
+    @Nullable
     @JsonIgnore
     public void removeAutomaticResults() {
         this.results = this.results.stream().filter(result -> result == null || !result.isAutomatic()).collect(Collectors.toList());
     }
 
+    /**
+     * removes all elements from the results list, which are null.
+     *
+     * This can be use to prepare a submission before sending it to the client. In some cases the submission is loaded from the database
```

Comment:
```suggestion
     * This can be uses to prepare a submission before sending it to the client. In some cases the submission is loaded from the database
```